### PR TITLE
Reuse pattern already in use for __has_attribute for __has_feature.

### PR DESCRIPTION
**Pull request by JasonLunn**

### CODE REVIEW

**Review by fowles — APPROVED** [review]

File: upb/port/undef.inc [hunk 1]
```diff
@@ -63,6 +63,7 @@
 #undef UPB_UNPOISON_MEMORY_REGION
 #undef UPB_ASAN
 #undef UPB_ASAN_GUARD_SIZE
+#undef UPB_HAS_FEATURE
```
Comment (fowles) [inline upb/port/undef.inc:66]: is UPB_HAS_ATTRIBUTE missing from this list?

Comment (JasonLunn) [reply upb/port/undef.inc:66]: Yes, though it is undef'd [here](https://github.com/protocolbuffers/upb/blob/3d0119a2dd674851107f35f7c039dfcc1b0d2ee4/upb/port/def.inc#L219).

Comment (fowles) [reply upb/port/undef.inc:66]: let's be consistent and undef these in the same place (your call which)

**Comment by acozzette** [comment]
I'm going to close this as we discussed offline, since we have moved upb development into the protobuf repo and this PR is not quite ready to merge as-is.

**Pull request closed without merging**